### PR TITLE
openssl: remove static libs from package + other fixes

### LINKED ISSUE #1030: [package] openssl/1.1.1d: building with mingw on Linux creates packages with both shared and static libs
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **openssl/1.1.1d**
  * Operating System+version: **Fedora Linux 30**
  * Compiler+version: **GCC 8**
  * Conan version: **conan 1.22.1**
  * Python version: **Python 3.7.6**

### Conan profile (output of `conan profile show default` or `conan profile show 
```
[settings]
os=Windows
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++11
compiler.threads=win32
compiler.exception=seh
build_type=Release
[options]
[build_requires]
[env]
AR=x86_64-w64-mingw32-ar
AS=x86_64-w64-mingw32-as
CC=x86_64-w64-mingw32-gcc
CXX=x86_64-w64-mingw32-g++
CPP=x86_64-w64-mingw32-cpp
LD=x86_64-w64-mingw32-ld
NM=x86_64-w64-mingw32-nm
OBJCOPY=x86_64-w64-mingw32-objcopy
OBJDUMP=x86_64-w64-mingw32-objdump
RC=x86_64-w64-mingw32-windres
STRIP=x86_64-w64-mingw32-strip
```

### Steps to reproduce (Include if Applicable)

`conan install openssl/1.1.1d@ -pr mingw`

The contents of the lib directory is:
```
├── lib
│   ├── engines-1_1
│   │   ├── capi.dll
│   │   └── padlock.dll
│   ├── libcrypto.a
│   ├── libcrypto.dll.a
│   ├── libssl.a
│   └── libssl.dll.a
```

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **openssl/all**

- separate build and package
- remove static libraries when building shared builds
- adhere to `no_zlib` option
- fixes to test_package
- move openssl/ALL -> openssl/all 

Fixes #1030 

Tested on Linux + mingw on linux

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openssl/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 2.8)
```
Comment (uilianries) [inline recipes/openssl/all/test_package/CMakeLists.txt]: You need to use `ALL` not `all`. One or two months ago I tried to rename to `all`, but for some reason, CCI didn't accept well.

Comment (uilianries) [reply recipes/openssl/all/test_package/CMakeLists.txt]: wait, is it working now?! Amazing! :clap:

Comment (madebr) [reply recipes/openssl/all/test_package/CMakeLists.txt]: The conversion also worked fine on Windows.
Some years ago, I remembered problems when changing cases.
Probably git fixed it.

**Comment by danimtb** [comment]
@madebr there are still some files under the ALL folder that should be moved to "all" like test_package. Please check it

**Comment by madebr** [comment]
> @madebr there are still some files under the ALL folder that should be moved to "all" like test_package. Please check it

I rebased on top of master.

**Comment by SSE4** [comment]
@madebr conflicts somehow appeared

**Comment by danimtb** [comment]
Yes, sorry @madebr. The changes are probably coming from #1170. Please merge master in this PR and it will be the next one merged, I promise 🙏

**Comment by madebr** [comment]
Because of the rename: `ALL` -> `all`.
I had to rebase again..

**Pull request merged**